controllers/taskcluster_builder.go

```diff
@@ -302,9 +305,14 @@ func (o *TaskClusterOperations) createDBUpgradeJob() []runtime.Object {
 		},
 	}
 
+	specBy, _ := json.Marshal(&o.dbUpgradeJob.Spec)
+	sum := sha256.Sum256(specBy)
+	o.dbUpgradeHash = base64.StdEncoding.EncodeToString(sum[:])
```

Comment:
worth putting a comment to explain why this hash is necessary? obviously it achieves something to do with ensuring the pod is recreated when it changes, but is there any further context required to say what issues you faced that require it, or is the intuition "this forces the pod to be recreated" good enough?